tests/unit/dataclasses/test_template_dc.py

```diff
@@ -220,3 +226,86 @@ def test_template_compilation(clean_building_motif):
     assert templ.name == "test"
     assert sorted(templ.parameters) == sorted(("name", "occ", "temp", "sp", "zone"))
     assert graph_size(templ.body) == 8
+
+
+def test_template_spreadsheet_generation(clean_building_motif):
+    lib = Library.load(directory="tests/unit/fixtures/templates")
+    BLDG = Namespace("urn:bldg/")
+
+    def add_spreadsheet_row(sheet, bindings):
+        num_columns = sheet.max_column
+        for column in range(1, num_columns + 1):
+            param = sheet.cell(row=1, column=column).value
+            sheet.cell(row=2, column=column).value = bindings[param][len(BLDG) :]
+
+    # test simple template, no deps, no optional, no inline
+    templ = lib.get_template_by_name("supply-fan")
+    bindings, filled = templ.fill(BLDG)
+    with NamedTemporaryFile(suffix=".xlsx") as dest:
+        templ.generate_spreadsheet(Path(dest.name))
+
+        w = openpyxl.load_workbook(dest.name)
+        add_spreadsheet_row(w.active, bindings)
+        w.save(Path(dest.name))
+
+        reader = XLSXIngress(Path(dest.name))
+        ing = TemplateIngress(templ, None, reader)
+        g = ing.graph(BLDG)
+
+        assert isomorphic(
+            g, filled
+        ), f"Template -> spreadsheet -> ingress -> graph path did not generate a result isomorphic to just filling the template {templ.name}"
+
+    # test simple template, no deps, WITH optional, no inline
```

Comment:
I think this should probably be a separate test

Reply:
Done!